### PR TITLE
Explicit authentication flow

### PR DESCRIPTION
**Pull request by KushalD1999**
**Work Done:**
Creates the handler to check if user is authenticated and a handler to request the token from extensionAuthHandler.

### CODE REVIEW

File: app/src/background/backgroundRequestHandler.ts [hunk 1]
```diff
@@ -111,3 +112,30 @@ BackgroundRequestHandler.on<backgroundRequest.FetchBreakpointRequestData>(
...
+ */
+BackgroundRequestHandler.on<backgroundRequest.GetAuthStateRequestData>(
+  backgroundRequest.BackgroundRequestType.IS_AUTHENTICATED,
+  async () => {
+    const request = await api.getAuthToken();
+    let response = false;
+    if (request !== "") {
+      response = true;
+    }
+    return response;
```
Comment (jamohile) [inline app/src/background/backgroundRequestHandler.ts:127]: The response type expects this to have format {isAuthenticated: true}

Comment (KushalD1999) [reply app/src/background/backgroundRequestHandler.ts:127]: Done

File: app/src/background/backgroundRequestHandler.ts [hunk 2]
```diff
@@ -111,3 +112,30 @@ BackgroundRequestHandler.on<backgroundRequest.FetchBreakpointRequestData>(
...
+  }
+);
+
+/**
+ * Handler for Request the token from extensionAuthHandler.
+ */
+BackgroundRequestHandler.on<backgroundRequest.AuthenticationRequestData>(
+  backgroundRequest.BackgroundRequestType.AUTHENTICATION,
+  async () => {
+    setInterval(() => {
```
[the remark this answers is not in the document]
Comment (KushalD1999) [reply app/src/background/backgroundRequestHandler.ts:205]: Fixed.

File: app/src/background/backgroundRequestHandler.ts [hunk 3]
```diff
@@ -111,3 +112,30 @@ BackgroundRequestHandler.on<backgroundRequest.FetchBreakpointRequestData>(
...
+      response = {isAuthenticated: true};
+    }
+    return response;
+  }
+);
+
+/**
+ * Handler for Request the token from extensionAuthHandler.
+ */
+BackgroundRequestHandler.on<backgroundRequest.AuthenticationRequestData>(
```
[the remark this answers is not in the document]
Comment (KushalD1999) [reply app/src/background/backgroundRequestHandler.ts:201]: Sg. Added response

**Pull request merged**